### ISSUE TITLE
-Wformat-signedness incorrectly warns when a bitfield field is passed to printf

### ISSUE THREAD

**Issue by ahatanak** [llvm/llvm-project#155665]
$ cat test.c
```
int printf(const char *, ...) __attribute__((__format__(__printf__, 1, 2)));

struct S {
  unsigned int bf : 16;
  unsigned char c;
};

struct S s;

void foo(void) {
  printf("%x\n", s.bf);
  printf("%x\n", s.c);
}
```

$ clang test.c -c -o -Wformat -Wformat-signedness 
test.c:11:18: warning: format specifies type 'unsigned int' but the argument has type 'int', which differs in signedness [-Wformat-signedness]
   11 |   printf("%x\n", s.bf);
      |           ~~     ^~~~
      |           %x
1 warning generated.

Both `s.bf` and `s.c` are promoted to `int` as a result of default argument promotion. clang silences the warning only when the latter is passed.
